File: hackhive-backend/src/main/java/com/hackhive/student/service/StudentNotificationPreferenceService.java
package com.hackhive.student.service;

import com.hackhive.student.dto.request.UpdateStudentNotificationPreferenceRequest;
import com.hackhive.student.dto.response.StudentNotificationPreferenceResponse;
import com.hackhive.student.entity.StudentNotificationPreference;
import com.hackhive.student.entity.StudentProfile;

public interface StudentNotificationPreferenceService {

    StudentNotificationPreferenceResponse getPreferences();

    StudentNotificationPreferenceResponse updatePreferences(UpdateStudentNotificationPreferenceRequest request);

    StudentNotificationPreference createDefaultPreferences(StudentProfile studentProfile);
}

File: hackhive-frontend/src/services/studentNotificationPreferenceService.js
import api from "../api/axios";

export const studentNotificationPreferenceService = {
    getPreferences: async () => {
        const response = await api.get("/student/notification-preferences");
        return response.data?.data ?? response.data;
    },

    updatePreferences: async (payload) => {
        const response = await api.put("/student/notification-preferences", payload);
        return response.data?.data ?? response.data;
    },
};

export default studentNotificationPreferenceService;
